bxsolana/provider/ws.ts

```diff
@@ -60,229 +60,428 @@ import {
     }
 
-    getOrderbook = (request: GetOrderbookRequest): Promise<GetOrderbookResponse> => {
-        return this.wsSocketCall("GetOrderbook", request)
+    async getOrderbook(request: GetOrderbookRequest): Promise<GetOrderbookResponse> {
+        return await this.wsConnection.call("GetOrderbook", request)
     }
 
-    getMarkets = (request: GetMarketsRequest): Promise<GetMarketsResponse> => {
-        return this.wsSocketCall("GetMarkets", request)
+    async getMarkets(request: GetMarketsRequest): Promise<GetMarketsResponse> {
+        return await this.wsConnection.call("GetMarkets", request)
     }
 
-    getTickers(request: GetTickersRequest): Promise<GetTickersResponse> {
-        return this.wsSocketCall("GetTickers", request)
+    async getTickers(request: GetTickersRequest): Promise<GetTickersResponse> {
+        return await this.wsConnection.call("GetTickers", request)
     }
 
-    getTrades(request: GetTradesRequest): Promise<GetTradesResponse> {
-        return this.wsSocketCall("GetTrades", request)
+    async getTrades(request: GetTradesRequest): Promise<GetTradesResponse> {
+        return await this.wsConnection.call("GetTrades", request)
     }
 
-    getServerTime(request: GetServerTimeRequest): Promise<GetServerTimeResponse> {
-        return this.wsSocketCall("GetServerTime", request)
+    async getServerTime(request: GetServerTimeRequest): Promise<GetServerTimeResponse> {
+        return await this.wsConnection.call("GetServerTime", request)
     }
 
-    getOpenOrders(request: GetOpenOrdersRequest): Promise<GetOpenOrdersResponse> {
-        return this.wsSocketCall("GetOpenOrders", request)
+    async getOpenOrders(request: GetOpenOrdersRequest): Promise<GetOpenOrdersResponse> {
+        return await this.wsConnection.call("GetOpenOrders", request)
     }
 
-    getUnsettled(request: GetUnsettledRequest): Promise<GetUnsettledResponse> {
-        return this.wsSocketCall("GetUnsettled", request)
+    async getUnsettled(request: GetUnsettledRequest): Promise<GetUnsettledResponse> {
+        return await this.wsConnection.call("GetUnsettled", request)
     }
 
-    getAccountBalance(request: GetAccountBalanceRequest): Promise<GetAccountBalanceResponse> {
-        return this.wsSocketCall("GetAccountBalance", request)
+    async getAccountBalance(request: GetAccountBalanceRequest): Promise<GetAccountBalanceResponse> {
+        return await this.wsConnection.call("GetAccountBalance", request)
     }
 
     //stream requests
-    getOrderbooksStream = (request: GetOrderbooksRequest): Promise<AsyncGenerator<GetOrderbooksStreamResponse>> => {
-        return this.wsSocketStreamCall("GetOrderbooksStream", request)
+    getOrderbooksStream = async (request: GetOrderbooksRequest): Promise<AsyncGenerator<GetOrderbooksStreamResponse>> => {
+        const subscriptionId = await this.wsConnection.subscribe("GetOrderbooksStream", request)
+
+        let count: number
+        if (this.streamToCountMap.size === 0) {
```

Comment:
Couple of things:
- this map looks like it will always grow in size, even after streams are closed. That's probably not a big deal, but the ideal implementation would not do that
- counter should probably be based off of `this.streamToCountMap.get("GetOrderbooksStream").size`? So you can cancel each index individual instead of having the counter increment across all streams?